Editor/EditorCore/MenuAction.cs

```diff
@@ -305,6 +319,23 @@ protected virtual void DoAlternateAction()
             MenuOption.Show(OnSettingsGUI, OnSettingsEnable, OnSettingsDisable);
         }
 
+#if UNITY_2023_2_OR_NEWER
+        /// <summary>
+        /// Replaces OnSettingsGUI for 2023.2 and newer to display settings in a SceneView overlay.
+        /// Creates a custom settings window for this action. Populate a root visual element in that method with
+        /// the settings content.
+        /// </summary>
+        protected internal virtual VisualElement CreateSettingsContent()
+        {
+            return null;
+        }
+
+        /// <summary>
+        /// If extra handles or gizmos are needed during the action execution in the scene, implement them here.
+        /// </summary>
+        protected internal virtual void DoSceneGUI(SceneView sceneView) {}
```

Comment:
Is `protected internal` the right choice for these? I imagine third party actions would need access to these.

Reply:
Good catch, I'll make it public then

Reply:
Same I guess for `CreateSettingsContent()`